eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/AbstractMutableList.java

```diff
@@ -608,63 +608,6 @@ public long injectInto(long injectedValue, LongObjectToLongFunction<? super T> f
         return SynchronizedMutableList.of(this);
     }
 
-    @Override
-    public MutableList<T> sortThis(Comparator<? super T> comparator)
-    {
-        if (this.size() < 10)
-        {
-            if (comparator == null)
-            {
-                this.insertionSort();
-            }
-            else
-            {
-                this.insertionSort(comparator);
-            }
-        }
-        else
-        {
-            this.defaultSort(comparator);
-        }
-        return this;
-    }
-
-    /**
-     * Override in subclasses where it can be optimized.
-     */
-    protected void defaultSort(Comparator<? super T> comparator)
-    {
-        Collections.sort(this, comparator);
-    }
-
-    private void insertionSort(Comparator<? super T> comparator)
-    {
-        for (int i = 0; i < this.size(); i++)
-        {
-            for (int j = i; j > 0 && comparator.compare(this.get(j - 1), this.get(j)) > 0; j--)
-            {
-                Collections.swap(this, j, j - 1);
-            }
-        }
-    }
-
-    private void insertionSort()
-    {
-        for (int i = 0; i < this.size(); i++)
-        {
-            for (int j = i; j > 0 && ((Comparable<T>) this.get(j - 1)).compareTo(this.get(j)) > 0; j--)
-            {
-                Collections.swap(this, j, j - 1);
-            }
-        }
-    }
-
-    @Override
-    public MutableList<T> sortThis()
-    {
-        return this.sortThis(Comparators.naturalOrder());
-    }
-
```

Comment:
Did this stuff move?

Reply:
Why is this stuff gone?

Reply:
I removed it.  There have been improvements to the `sort` implementation in the JDK with the addition of `TimSort`.  I don't think this code pulls its weight since we started compiling with Java 8.  Now that I am overriding sort in all of the `AbstractMemoryEfficientMutableList` subclasses, this code is even less beneficial.

If you think it is a mistake this remove this optimization, let's discuss how we should add it back.  It should probably move from `sortThis` to `sort`.

Reply:
I'm looking to move this code down to `AbstractMemoryEfficientMutableList` removing the size check and deleting all of the current specialized implementations in the subclasses.  Make sense?

Reply:
Not trying to encourage a particular optimization, quite the opposite. I find this stuff hard to review. The most important issue is that without overriding sort() and replaceAll(), behavior is wrong on thread-safe collections. So I came into this expecting additional overrides.